src/Aspire.Hosting.Azure/Extensions/AzureRedisExtensions.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Aspire.Hosting.Azure.Redis;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding the Azure Redis resources to the application model.
+/// </summary>
+public static class AzureRedisExtensions
+{
+    /// <summary>
+    /// Publishes the Azure Redis resource to the manifest.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="callback">The callback to configure the underlying Azure resource.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<RedisResource> PublishAsAzureRedis(this IResourceBuilder<RedisResource> builder, Action<AzureRedisResource>? callback = null)
```

Comment:
Why is the callback needed you think? The user already has the RedisResource.

Reply:
Oh wait this is the AzureRedisResource. Got it! It might make sense for this to be `IResourceBuilder<AzureRedisResoruce>`. That's where all the extension methods would be.

Reply:
Yeah good point. In fact that is why we have the resource builder 😅